pkg/artifacts/signable.go

```diff
@@ -149,40 +152,56 @@ type image struct {
 }
 
 func (oa *OCIArtifact) ExtractObjects(ctx context.Context, obj objects.TektonObject) []interface{} {
-	log := logging.FromContext(ctx)
 	objs := []interface{}{}
-
-	// TODO: Not applicable to PipelineRuns, should look into a better way to separate this out
-	if tr, ok := obj.GetObject().(*v1beta1.TaskRun); ok {
-		imageResourceNames := map[string]*image{}
-		if tr.Status.TaskSpec != nil && tr.Status.TaskSpec.Resources != nil {
-			for _, output := range tr.Status.TaskSpec.Resources.Outputs {
-				if output.Type == backport.PipelineResourceTypeImage {
-					imageResourceNames[output.Name] = &image{}
-				}
+	if trV1, ok := obj.GetObject().(*v1.TaskRun); ok {
+		var resources v1beta1.TaskResources //nolint:staticcheck
+		shouldReplace := false
+		if serializedResources, ok := trV1.Annotations["tekton.dev/v1beta1-status-taskrunstatusfields-taskspec-resources"]; ok {
+			if err := json.Unmarshal([]byte(serializedResources), &resources); err == nil {
+				shouldReplace = true
 			}
 		}
-
-		for _, rr := range tr.Status.ResourcesResult {
-			img, ok := imageResourceNames[rr.ResourceName]
-			if !ok {
-				continue
-			}
-			// We have a result for an image!
-			if rr.Key == "url" {
-				img.url = rr.Value
-			} else if rr.Key == "digest" {
-				img.digest = rr.Value
+		var results []v1beta1.RunResult //nolint:staticcheck
+		if serializedResources, ok := trV1.Annotations["tekton.dev/v1beta1ResourcesResult"]; ok {
+			if err := json.Unmarshal([]byte(serializedResources), &results); err == nil {
+				shouldReplace = shouldReplace && true
 			}
 		}
+		trV1Beta1 := &v1beta1.TaskRun{} //nolint:staticcheck
+		if err := trV1Beta1.ConvertFrom(ctx, trV1); err == nil {
+			if shouldReplace {
+				trV1Beta1.Status.TaskSpec.Resources = &resources //nolint:staticcheck
+				trV1Beta1.Status.ResourcesResult = results       //nolint:staticcheck
+			}
+			imageResourceNames := map[string]*image{}
+			if trV1Beta1.Status.TaskSpec != nil && trV1Beta1.Status.TaskSpec.Resources != nil { //nolint:staticcheck
+				for _, output := range trV1Beta1.Status.TaskSpec.Resources.Outputs { //nolint:staticcheck
```

Comment:
Thanks for the offline discussion. I think what's pointed is mostly accurate here.
- [The conversion func only converts resources field in TaskSpec](https://github.com/tektoncd/pipeline/blob/e8d3bfdc03fdd7fa2b4bce37301ff558aedb0e1b/pkg/apis/pipeline/v1beta1/task_conversion.go#L330)
- [The outputs in TaskRunStatus is not handled](https://github.com/tektoncd/pipeline/blob/e8d3bfdc03fdd7fa2b4bce37301ff558aedb0e1b/pkg/apis/pipeline/v1beta1/taskrun_conversion.go#L231)

That said only the resources fields are preserved, the `trV1Beta1.Status.TaskSpec.Resources.Outputs` will likely to be nil and this might be error out.

Moving forward we might either add back conversion function for resources in TaskRunStatus in pipeline or add a nil guard to this (though not too sure if there is going to any use cases).

Reply:
IIUC, pipeline conversion funcs aren't complete, so there are extra steps we need to take here to ensure compatibility?

Should we be adding this back upstream to pipelines instead then? 🤔

Reply:
I think this sounds like the way to go for keeping the backwards compatibility.
Raised https://github.com/tektoncd/pipeline/issues/7504

Reply:
With https://github.com/tektoncd/pipeline/pull/7507 merged and this PR using the updated libs I believe this thread is resolved.  Markind resolved